app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx

```diff
@@ -582,23 +614,57 @@ interface TerminalRunProps {
   runStatus: RunStatus | null
   handleClearClick: () => void
   isClosingCurrentRun: boolean
+  setShowRunFailedModal: (showRunFailedModal: boolean) => void
+  highestPriorityError?: RunError
 }
 function TerminalRunBanner(props: TerminalRunProps): JSX.Element | null {
-  const { runStatus, handleClearClick, isClosingCurrentRun } = props
+  const {
+    runStatus,
+    handleClearClick,
+    isClosingCurrentRun,
+    setShowRunFailedModal,
+    highestPriorityError,
+  } = props
   const { t } = useTranslation('run_details')
+
+  const handleClick = (): void => {
+    handleClearClick()
+    setShowRunFailedModal(true)
+  }
+
   if (runStatus === RUN_STATUS_FAILED || runStatus === RUN_STATUS_SUCCEEDED) {
     return (
-      <Banner
-        type={runStatus === RUN_STATUS_FAILED ? 'error' : 'success'}
-        onCloseClick={handleClearClick}
-        isCloseActionLoading={isClosingCurrentRun}
-      >
-        <Flex justifyContent={JUSTIFY_SPACE_BETWEEN} width="100%">
-          {runStatus === RUN_STATUS_FAILED
-            ? t('run_failed')
-            : t('run_completed')}
-        </Flex>
-      </Banner>
+      <>
+        {runStatus === RUN_STATUS_SUCCEEDED ? (
```

Comment:
do you need this empty fragment?

```suggestion
 return runStatus === RUN_STATUS_SUCCEEDED ? (
```

Reply:
yeah, before we could use <Banner> to wrap children since we just needed to close the banner itself.
but now for success we need to close banner and for failed, we need to open the modal and change the layout a little bit. I just avoid using multiple conditions inside Banner.